src/fs/mounts/macos.rs

```diff
@@ -0,0 +1,43 @@
+use std::{mem, ptr};
+use std::ffi::{CStr, OsStr};
+use std::os::raw::{c_char, c_int};
+use std::os::unix::ffi::OsStrExt;
+use std::path::PathBuf;
+use libc::{__error, getfsstat, MNT_NOWAIT, statfs};
+use crate::fs::mounts::{Error, MountedFs};
+
+/// Get a list of all mounted filesystem
+pub fn mounts() -> Result<Vec<MountedFs>, Error> {
+    // Passing a null pointer and zero bufsize will return the number of mounts
+    let mut count: i32 = unsafe { getfsstat(ptr::null_mut(), 0, MNT_NOWAIT) };
```

Comment:
If the `unsafe` is unavoidable please can you add SAFETY comments as per https://rust-lang.github.io/rust-clippy/master/#/undocumented_unsafe_blocks?